### PR TITLE
Resolve lint & test errors and integrate Travis CI

### PR DESCRIPTION
**Pull request by kadamwhite**
- Resolves lint errors in the master branch
- Resolves unit test failures in the master branch
- Adds `.travis.yml` file to add continuous integration via travis-ci.org

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "server:prod": "node backend/server",
     "server:dev": "supervisor -w backend/ backend/server",
     "lint": "eslint --ext js,jsx frontend",
-    "test": "node jest/run.js"
+    "test": "node jest/run.js",
+    "ci": "npm test && npm run lint"
```
Comment (kadamwhite) [inline package.json]: @gnarf I'd appreciate your feedback on what combination and ordering of commands "feels best" for CI integration

File: frontend/components/PerceivedDemographics.jsx [hunk 2]
```diff
@@ -75,18 +92,20 @@ class PerceivedDemographicsContainer extends Component {
...
+PerceivedDemographicsContainer.propTypes = {
+  onEnter: PropTypes.func.isRequired,
+};
 
 const mapDispatchToProps = dispatch => ({
   onEnter: () => dispatch(requestFaces()),
 });
 
-export default connect(mapStateToProps, mapDispatchToProps)(PerceivedDemographicsContainer);
+export default connect(null, mapDispatchToProps)(PerceivedDemographicsContainer);
```
Comment (kadamwhite) [inline frontend/components/PerceivedDemographics.jsx]: mapStateToProps was returning an empty object, so passing a falsy value here lets us skip having to declare the method at all.

File: frontend/components/PerceivedDemographics.jsx [hunk 3]
```diff
@@ -75,18 +92,20 @@ class PerceivedDemographicsContainer extends Component {
...
       </div>
-    )
+    );
   }
 }
 
-const mapStateToProps = state => ({});
+PerceivedDemographicsContainer.propTypes = {
+  onEnter: PropTypes.func.isRequired,
+};
```
Comment (kadamwhite) [inline frontend/components/PerceivedDemographics.jsx]: PropTypes allows us to enforce the values that are being passed to our component are of the proper type

File: frontend/components/PerceivedDemographics.jsx [hunk 4]
```diff
@@ -1,72 +1,89 @@
...
               <option value="genderqueer">genderqueer</option>
             </select>
           </label>
 
-          <label>
+          <label htmlFor="perceivedEthnicity">
             perceived ethnicity:
-            <select name="percievedEthnicity" value={this.state.percievedEthnicity} onChange={this.handleInputChange}>
+            <select
+              id="perceivedEthnicity"
```
Comment (kadamwhite) [inline frontend/components/PerceivedDemographics.jsx]: I'm not sure to what extent this is strictly _required_ given the nesting of the control within the label, but our a11y linting rules recommended adding a unique ID to each form element and explicitly linking that ID with `htmlFor=` in the parent label element.

**Comment by boazsender** [comment]
+1 runs on my machine, merging.

**Pull request merged**